### PR TITLE
typeahead-nf: wire-in new search api

### PR DESCRIPTION
**Pull request by raymondng**
this should improve efficiency & responsiveness of typeahead endpoint

cc: @yingjieMiao

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala [hunk 1]
```diff
@@ -347,13 +347,10 @@ class TypeaheadCommander @Inject()(
             } else None
             ConnectionWithInviteStatus(u.fullName, hit.score, SocialNetworks.FORTYTWO.name, picUrl, s"fortytwo/${u.externalId}", "joined")
           case SocialNetworks.FORTYTWO_NF =>
-            val bu = hit.info.asInstanceOf[BasicUser] // todo: uptake User API from search
+            val bu = hit.info.asInstanceOf[BasicUserWithUserId] // todo: uptake User API from search
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala:350]: is this type cast necessary?

**Comment by yingjieMiao** [comment]
should save some milliseconds ;-)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala:350]: yes for current code -- which is not ideal in this aspect. I think I have a todo somewhere to track this ...

**Comment by raymondng** [comment]
yup that's the goal :)

**Pull request merged**